ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/core/TransactionPool.java

```diff
@@ -49,17 +49,20 @@
   private final ProtocolSchedule<?> protocolSchedule;
   private final ProtocolContext<?> protocolContext;
   private final TransactionBatchAddedListener transactionBatchAddedListener;
+  private final Synchronizer synchronizer;
   private Optional<AccountFilter> accountFilter = Optional.empty();
 
   public TransactionPool(
       final PendingTransactions pendingTransactions,
       final ProtocolSchedule<?> protocolSchedule,
       final ProtocolContext<?> protocolContext,
-      final TransactionBatchAddedListener transactionBatchAddedListener) {
+      final TransactionBatchAddedListener transactionBatchAddedListener,
+      final Synchronizer synchronizer) {
     this.pendingTransactions = pendingTransactions;
     this.protocolSchedule = protocolSchedule;
     this.protocolContext = protocolContext;
     this.transactionBatchAddedListener = transactionBatchAddedListener;
+    this.synchronizer = synchronizer;
```

Comment:
Can we just pass around a shared SyncStatus object instead of the whole synchronizer?

Reply:
This is not a constructive review comment. Obviously we can do that -- do you think we ought to do that? If that's what you're suggesting, why do you feel so? Why would that be better than the current paradigm?

Reply:
That was the implied subtext.  Let me try the direct approach.

I don't think the entire synchronizer object should be passed around when all we need is the sync state.  Change it and pass the SyncState object instead unless there is a technical reason we cannot.

Reply:
So to do this right we need to change how we are handling the SyncStatus object.  Currently it is an Optional and not returned when we are in sync.

I propose we make it always return SyncStatus, not an Optional<SyncStatus>.  We also add a field that maps the three current states the sync can be in:  NO_STARTED, SYNCING, and UP_TO_DATE.  NOT_STARTED corresponds to a highest block of zero, and up to date corresponds to current block equal to highest block.  We can also add a method to tell us how many blocks behind we are and use that in the logic where we would drop inbound transactions.

This object is shared between the synchronizer and the transaction pool.  Synchronizer writes to it and tx pool reads.

Reply:
You want to share `SyncState` the same way the `AbstractMiningCoordinator` does.  `SyncStatus` is not what you want - it's only for the JSON-RPC side of things (we should, separate to this work, push it to be just in the JSON-RPC realm and not in `Synchronizer` at all).